### ISSUE TITLE
Consider moving chat to Gitter

### ISSUE THREAD

**Issue by captbaritone** [divad12/vim-awesome#85]
A number of open source projects now use Gitter as their chat hangout over the more traditional IRC. I'm not crazy about moving from an open platform to a closed one but I think it may be worth the ideological sacrifice. I'll list some pros and cons:

## Pros

1. Easy access to chat history from when you were away
2. Easy access on mobile (their app is pretty good)
3. We could probably move our existing HipChat integration to use their API
4. Uses our existing GitHub permissions
5. Lower barrier to entry
6. I personally already hang out in other Gitters ;)

## Cons

1. Gitter is a closed platform :disappointed: 

Thoughts? Especially @jonafato @spicyj and @divad12

**Comment by Shougo** [comment]
:+1:

**Comment by jonafato** [comment]
TL;DR: I'm fine with IRC, but I won't argue against Gitter much either.

Posting the comments I made in IRC (with some additions and modifications) here for better visibility:

The last couple of times I used Gitter, I wasn't a big fan. Overall, I found the app to be kind of sluggish (particularly when navigating between channels). Things may have improved since then, so I can't really speak to the current state of the app. As @captbaritone  notes above, the lack of openness is also a concern of mine.

That said, I recognize that it has a lower barrier to entry than IRC for many people, and there aren't a ton of other easy options. I'm optimistic about the direction of https://matrix.org/, and I like the concept of Dropbox's https://www.zulip.org/, but I haven't used either enough to recommend them as an alternative. I'm not opposed to using gitter in any principled sense, though, so I'm happy to lurk there if requested.

Ultimately, based on experience from other projects, very few people outside of core contributors will use the chat room. This means that it's best to pick whatever solution fits into the core team's workflow best and optimize for other factors secondarily and once they become relevant (i.e. avoid premature optimization). IRC is part of my normal development workflow (including ZNC to avoid missing messages), so I'm not itching to move off of it, and one extra channel is minimal effort for me. For others, this may not be the case, and it's probably easier to get started with Gitter than IRC. Ease of getting started is a really important factor here; if no one is going to put out the effort to sign into chat, all other arguments become moot.

**Comment by captbaritone** [comment]
@jonafato Thanks for your thoughts. All valid!

I've updated the Readme and made Gitter our main hangout with the understanding that if it fails to meet our needs we can easily move somewhere else.

https://github.com/divad12/vim-awesome/commit/d440d9970450b4def6c560d28d7094e373377211

Join me! https://gitter.im/divad12/vim-awesome

**Comment by divad12** [comment]
Great, thanks for setting that up @captbaritone! Let's see how it goes.

I'm open to either — I don't use either right now! Been too busy with other things to do much in open source land. And I'm handing this off to you and others who can spend the proper time and care to contribute and maintain to this project, so I leave it up to y'all!
